spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigMapPropertySourceLocator.java

```diff
@@ -59,10 +67,28 @@ protected MapPropertySource getMapPropertySource(String name,
 			ConfigMapConfigProperties.NormalizedSource normalizedSource, String configurationTarget,
 			ConfigurableEnvironment environment) {
 
-		String fallbackNamespace = kubernetesNamespaceProvider != null ? kubernetesNamespaceProvider.getNamespace()
-				: kubernetesClientProperties.getNamespace();
-		return new KubernetesClientConfigMapPropertySource(coreV1Api, name,
-				getNamespace(normalizedSource, fallbackNamespace), environment, normalizedSource.getPrefix());
```

Comment:
this is probably the main idea of this PR. A few obvious things first. We used to do : 

 - find `fallbackNamespace`; either from `kubernetesNamespaceProvider` or `kubernetesClientProperties`.
 - if `NormalizedSource::getNamespace` was present, use that, otherwise use `fallbackNamespace`.

In the vast majority of cases `NormalizedSource::getNamespace` is going to be present so the logic of the above should be inverted. This way we might not even need to call `kubernetesNamespaceProvider` or `kubernetesClientProperties`.

Now let's look at what used to happen when someone used the deprecated constructor that used `kubernetesClientProperties`.  It is actually pretty trivial, namespace was computed as either (in this order):

- `NormalizedSource::getNamespace`
- `kubernetesClientProperties.getNamespace()`

If you look at what I did in the refactor, I did the same thing. First:


```
if (StringUtils.hasText(normalizedNamespace)) {
			namespace = normalizedNamespace;
}
```

and then:

```
            if (StringUtils.hasText(kubernetesClientProperties.getNamespace())) {
				namespace = kubernetesClientProperties.getNamespace();
			}
			else {
				throw new NamespaceResolutionFailedException(
						"could not resolve namespace in normalized source or KubernetesClientProperties");
			}
```

Throwing an Exception in this case is safe. Why? Because in the previous case, failure to provide a namespace would have resulted in an Exception _anyway_, simply because later we call the `CoreV1Api` that uses this namespace. 

The last branch here: 


```
namespace = KubernetesClientConfigUtils.getApplicationNamespace(normalizedNamespace, "Config Map",
					kubernetesNamespaceProvider);
```

simply gets the namespace from either (in this order): 

- `NormalizedSource::getNamespace`
- `kubernetesClientProvider.getNamespace()`

via `KubernetesClientConfigUtils::getApplicationNamespace`. Internally this method will throw `NamespaceResolutionFailedException` too if no namespace is found.